crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -122,26 +122,24 @@ impl ProtocolFees {
         {
             vec![]
         } else {
-            self
-                    .fee_policies
-                    .iter()
-                    // TODO: support multiple fee policies
-                    .find_map(|fee_policy| {
-                        let outside_market_price = boundary::is_order_outside_market_price(&order_, &quote_, order.data.kind);
-                        match (outside_market_price, &fee_policy.order_class) {
-                            (_, OrderClass::Any) => Some(&fee_policy.policy),
-                            (true, OrderClass::Limit) => Some(&fee_policy.policy),
-                            (false, OrderClass::Market) => Some(&fee_policy.policy),
-                            _ => None,
-                        }
-                    })
-                    .and_then(|policy| match policy {
-                        policy::Policy::Surplus(variant) => variant.apply(&order),
-                        policy::Policy::PriceImprovement(variant) => variant.apply(&order, quote),
-                        policy::Policy::Volume(variant) => variant.apply(&order),
-                    })
-                    .into_iter()
-                    .collect_vec()
+            self.fee_policies
```

Comment:
There is currently still an early return for the partner fee. AFAIU supporting multiple fees is specifically intended to allow setting a partner fee alongside a regular protocol fee.

Reply:
Aha, ok, I thought partner fee was still "exclusive", got it! thanks!

Reply:
style nit: by now it probably makes sense to have a separate early return for the exempt case right after computing the protocol fee. That way you get rid of the indentation of the `else` branch and can move the `quote_` stuff closer to where it's needed.
```
let partner_fee = //...
if exempt {
    return partner_fee;
}
let protocol_fees = //...
protocol_fees.into_iter().chain(partner_fee).collect()
```